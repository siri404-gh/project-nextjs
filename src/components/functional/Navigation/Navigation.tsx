import React, { Fragment, FunctionComponent } from 'react';
import {
  Stepper,
  Step,
  StepLabel,
  makeStyles,
  Link as _Link,
} from '@material-ui/core';
import Link from 'next/link';
import cn from 'classnames';

const useStyles = makeStyles(theme => ({
  root: {
    clear: 'both',
    background: theme.palette.background.default,
    padding: 8,
    borderRadius: 10,
  },
  leftLink: {
    textAlign: 'left',
    display: 'inline-block',
    textDecoration: 'none',
    cursor: 'pointer',
  },
  rightLink: {
    textAlign: 'right',
    display: 'inline-block',
    textDecoration: 'none',
    cursor: 'pointer',
  },
}));

export interface Props {
  className?: string,
  prev?: linkType,
  next?: linkType,
}

export type linkType = {
  title: string,
  route: string,
};

const Navigation: FunctionComponent<Props> = ({ className, prev, next }) => {
  const classes = useStyles();

  return <Fragment>
    <Stepper activeStep={-1} className={cn(classes.root, className)}>
      <Step>
        <StepLabel icon={null}>
          {prev
            ? <_Link href={prev.route} className={classes.leftLink}>{prev.title}</_Link>
            : <_Link href={'/'}>Home</_Link>
          }
        </StepLabel>
      </Step>
      <Step>
        <StepLabel icon={null}>
          {next
            ? <_Link href={next.route} className={classes.leftLink}>{next.title}</_Link>
            : <_Link href={'/'}>Home</_Link>
          }
        </StepLabel>
      </Step>
    </Stepper>
  </Fragment>
};

Navigation.defaultProps = {
  className: '',
  prev: null,
  next: null,
};

export default Navigation;
