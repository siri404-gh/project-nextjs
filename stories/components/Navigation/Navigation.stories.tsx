import React from 'react';
import { Story } from '@storybook/react/types-6-0';

import Navigation, { Props } from '@/components/functional/Navigation/Navigation';

export default {
  title: 'Navigation',
  component: Navigation,
};

const Template: Story<Props> = args => <Navigation {...args} />;

export const FirstStory = Template.bind({});

FirstStory.args = {
  next: {
    title: 'next',
    route: '/route',
  },
  prev: {
    title: 'prev',
    route: '/route',
  },
};
