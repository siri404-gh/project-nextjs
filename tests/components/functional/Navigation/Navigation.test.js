import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';

import Navigation from '@/components/functional/Navigation/Navigation';

describe('Navigation', () => {
  let container;

  beforeEach(() => {
    container = document.createElement('section');
    document.body.appendChild(container);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it('Should render without props', () => {
    act(() => {
      render(<Navigation />, container);
    });
    expect(container).toMatchSnapshot();
  });

  it('Should render with props', () => {
    act(() => {
      const props = {
        next: {
          title: 'title',
          route: '/route',
        },
        prev: {
          title: 'title',
          route: '/route',
        },
      };
      render(<Navigation {...props} />, container);
    });
    expect(container).toMatchSnapshot();
  });
});
