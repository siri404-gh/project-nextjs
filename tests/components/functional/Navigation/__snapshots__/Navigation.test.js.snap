// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Should render with props 1`] = `
<section>
  <div
    class="MuiPaper-root MuiStepper-root MuiStepper-horizontal makeStyles-root-4 MuiPaper-elevation0"
  >
    <div
      class="MuiStep-root MuiStep-horizontal"
    >
      <span
        class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
      >
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
          >
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-leftLink-5 MuiTypography-colorPrimary"
              href="/route"
            >
              title
            </a>
          </span>
        </span>
      </span>
    </div>
    <div
      class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
    >
      <span
        class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
      />
    </div>
    <div
      class="MuiStep-root MuiStep-horizontal"
    >
      <span
        class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
      >
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
          >
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-leftLink-5 MuiTypography-colorPrimary"
              href="/route"
            >
              title
            </a>
          </span>
        </span>
      </span>
    </div>
  </div>
</section>
`;

exports[`Navigation Should render without props 1`] = `
<section>
  <div
    class="MuiPaper-root MuiStepper-root MuiStepper-horizontal makeStyles-root-1 MuiPaper-elevation0"
  >
    <div
      class="MuiStep-root MuiStep-horizontal"
    >
      <span
        class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
      >
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
          >
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              href="/"
            >
              Home
            </a>
          </span>
        </span>
      </span>
    </div>
    <div
      class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
    >
      <span
        class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
      />
    </div>
    <div
      class="MuiStep-root MuiStep-horizontal"
    >
      <span
        class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
      >
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
          >
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              href="/"
            >
              Home
            </a>
          </span>
        </span>
      </span>
    </div>
  </div>
</section>
`;
